Assets/Scripts/Player/Body/BodyPart.cs

```diff
@@ -0,0 +1,101 @@
+using System.Collections.Generic;
+using Enums;
+using UnityEngine;
+
+namespace Player.Body
+{
+    /// <summary>
+    /// Class is responsible for handling state changes associated with a particular body part.
+    /// MonoBehaviour should be attached to the game object that contains the hitbox collider of the particular body part.
+    /// Each body part with its own collider should also have an instance of this component.
+    /// </summary>
+    public class BodyPart : MonoBehaviour
+    {
+        //Specifies which body part this is
+        [SerializeField] private BodyPartType bodyPartType;
+        //Specifies prefab to spawn if this body part is detached
+        [SerializeField] private GameObject severedBodyPartPrefab;
+        //List of children for this bodypart. For example, the hand should be a child of the arm, etc.
+        [SerializeField] private List<BodyPart> childrenParts;
+        //Flag enum, storing the current active statuses for this bodypart
+        [SerializeField] private BodyPartStatuses bodyPartStatuses;
+        //The skinnedMeshRenderer associated with this bodypart. It will be hidden if the bodypart is detached
+        [SerializeField] private SkinnedMeshRenderer skinnedMeshRenderer;
+        //The body component that this bodypart belongs to
+        [SerializeField] private Body body;
+
+        //The types and amounts of damages that this bodypart has suffered
+        private List<BodyPartDamage> bodyPartDamages = new List<BodyPartDamage>
+        {
+            new BodyPartDamage(DamageType.Burn, 0f), 
+            new BodyPartDamage(DamageType.Brute, 0f),
+            new BodyPartDamage(DamageType.Toxic, 0f), 
+            new BodyPartDamage(DamageType.Suffocation, 0f)
+        };
+
+        public BodyPartType BodyPartType => bodyPartType;
+
+        public GameObject SeveredBodyPartPrefab => severedBodyPartPrefab;
+
+        public List<BodyPart> ChildrenParts => childrenParts;
+
+        public BodyPartStatuses BodyPartStatuses => bodyPartStatuses;
+
+        public List<BodyPartDamage> BodyPartDamages => bodyPartDamages;
+
+        public SkinnedMeshRenderer SkinnedMeshRenderer => skinnedMeshRenderer;
+
```

Comment:
These linebreaks are excessive. Removing them would group the properties together for better readability